src/js_realm.hpp

```diff
@@ -689,6 +689,33 @@ bool RealmClass<T>::get_realm_config(ContextType ctx, size_t argc, const ValueTy
                 config.should_compact_on_launch_function = std::move(should_compact_on_launch_functor);
             }
 
+            static const String data_initalization_string = "dataInitialization";
+            ValueType data_initialization_value = Object::get_property(ctx, object, data_initalization_string);
+            if (!Value::is_undefined(ctx, data_initialization_value)) {
+                if (config.schema_mode == SchemaMode::Immutable) {
+                    throw std::invalid_argument("Cannot set 'dataInitalization' when 'readOnly' is set.");
+                }
+
+                FunctionType data_initialization_function =
+                    Value::validated_to_function(ctx, data_initialization_value);
+                config.initialization_function = [=](SharedRealm realm) {
+
+                    auto realm_ptr = new SharedRealm(realm);
```

Comment:
Why allocate a new shared Realm on the heap when one is passed as argument to the lambda?

Reply:
I am calling `reset()` to release the ownership.

Reply:
But won't realm_ptr be a pointer to a heap allocated `std::shared_ptr<Realm>`? I.e. do you actually free the memory allocated for the smartpointer itself by calling `reset` on it?

Anyways - why can't you use the `realm` passed as argument to the lambda directly?

Reply:
I agree, it seems like `realm_ptr` would stick around and not be de-allocated.
And, because `realm` is passed as copy-by-value, you should have a fresh `shared_ptr` ready to go there.

Reply:
@fronck @kraenhansen I have slightly modified the code, and I believe I have addressed your concerns.

Reply:
Okay, I tried passing `&realm` but get a double free. Since `create_object` calls `set_internal` it does assume ownership of the pointer and therefore needs a brand new internal object. I can't say I love an API that implicitly assumes ownership like that.